### PR TITLE
Update address form

### PR DESCRIPTION
**Pull request by dreampulse**
resolves https://github.com/all3dp/printing-engine-client/issues/1131

### CODE REVIEW

File: src/app/component/country-select-field/country-select-field.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  const renderReadMode = () => (
+    <div className="countrySelectField__readMode">
+      <div>{value ? getCountryName(value) : '-'}</div>
+      <Button text tiny label={changeButtonLabel} onClick={() => setEditMode(true)} />
+    </div>
+  )
+
+  return (
+    <div className={cn('countrySelectField', {}, classNames)}>
```
Comment (ranzwertig) [inline src/app/component/country-select-field/country-select-field.js]:
```suggestion
    <div className={cn('CountrySelectField', {}, classNames)}>
```

File: src/app/component/country-select-field/country-select-field.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        <div className="countrySelectField__notification">
+          {initialValue !== value ? changedLabel : changeLabel}
+        </div>
+      }
+    </>
+  )
+
+  const renderReadMode = () => (
+    <div className="countrySelectField__readMode">
+      <div>{value ? getCountryName(value) : '-'}</div>
```
Comment (ranzwertig) [inline src/app/component/country-select-field/country-select-field.js]:
```suggestion
      {value ? getCountryName(value) : '-'}
```

File: src/app/component/input-field/_input-field.scss [hunk 3]
```diff
@@ -1,4 +1,4 @@
-.input-field {
+.inputField {
```
Comment (ranzwertig) [inline src/app/component/input-field/_input-field.scss]:
```suggestion
.InputField {
```

File: src/app/component/input-field/input-field.js [hunk 4]
```diff
@@ -29,24 +30,31 @@ export default class InputField extends Component {
...
     const inputId = id || uniqueId('uid-')
-    const finalModifiers = [
-      ...modifiers,
-      {empty: !this.props.value},
-      {disabled: this.props.disabled}
-    ]
+
     return (
-      <div className={buildClassName('input-field', finalModifiers, classNames)}>
+      <div className={cn('inputField', {error, warning, disabled, empty: !value}, classNames)}>
```
Comment (ranzwertig) [inline src/app/component/input-field/input-field.js]:
```suggestion
      <div className={cn('InputField', {error, warning, disabled, empty: !value}, classNames)}>
```

File: src/app/component/labeled-checkbox/_labeled-checkbox.scss [hunk 5]
```diff
@@ -1,4 +1,4 @@
-.labeled-checkbox {
+.labeledCheckbox {
```
Comment (ranzwertig) [inline src/app/component/labeled-checkbox/_labeled-checkbox.scss]:
```suggestion
.LabeledCheckbox {
```

File: src/app/component/labeled-checkbox/labeled-checkbox.js [hunk 6]
```diff
@@ -2,29 +2,31 @@ import PropTypes from 'prop-types'
...
   label,
+  error = false,
+  warning = false,
   onChange = () => {},
   name = ''
 }) => {
   const id = uniqueId('uid-')
   return (
-    <div className={buildClassName('labeled-checkbox', modifiers, classNames)}>
+    <div className={cn('labeledCheckbox', {}, classNames)}>
```
Comment (ranzwertig) [inline src/app/component/labeled-checkbox/labeled-checkbox.js]:
```suggestion
    <div className={cn('LabeledCheckbox', {}, classNames)}>
```

File: src/app/component/select-field/_select-field.scss [hunk 7]
```diff
@@ -1,4 +1,4 @@
-.select-field {
+.selectField {
```
Comment (ranzwertig) [inline src/app/component/select-field/_select-field.scss]:
```suggestion
.SelectField {
```

File: src/app/component/select-field/select-field.js [hunk 8]
```diff
@@ -62,26 +65,38 @@ export default class SelectField extends Component {
...
-      }
-    ]
 
     const renderButton = (portal = {}) => (
       <button
         type="button"
         key="button"
-        className={buildClassName('select-field', finalModifiers, classNames)}
+        className={cn(
+          'selectField',
```
Comment (ranzwertig) [inline src/app/component/select-field/select-field.js]:
```suggestion
          'SelectField',
```

File: src/app/container/modal/address-form.js [hunk 9]
```diff
@@ -294,20 +252,39 @@ const AddressFormModal = ({
...
+            <Field
+              validate={phoneNumber}
+              component={renderFormikField(InputField)}
+              label="Phone number"
+              name="phoneNumber"
+              type="tel"
+            />
+          </FormRow>
+
+          <Field
```
Comment (ranzwertig) [inline src/app/container/modal/address-form.js]: This field is outside a FormRow

File: src/app/lib/validator.spec.js [hunk 10]
```diff
@@ -26,3 +26,14 @@ describe('vat()', () => {
   it('returns undefined if value is empty and it is a eu country', () =>
     expect(vat('DE', ['US'])(''), 'to be undefined'))
 })
+
+describe('phoneNumber()', () => {
+  it('returns undefined if a string is provide', () =>
```
Comment (ranzwertig) [inline src/app/lib/validator.spec.js]:
```suggestion
  it('returns undefined if a valid string is provided', () =>
```

File: src/app/lib/validator.spec.js [hunk 11]
```diff
@@ -26,3 +26,14 @@ describe('vat()', () => {
...
+    expect(phoneNumber('123'), 'to be undefined'))
+
+  it('returns "Invalid phone number" if an empty string is provide', () =>
```
Comment (ranzwertig) [inline src/app/lib/validator.spec.js]:
```suggestion
  it('returns "Invalid phone number" if an empty string is provided', () =>
```

File: src/app/lib/validator.spec.js [hunk 12]
```diff
@@ -26,3 +26,14 @@ describe('vat()', () => {
...
+    expect(phoneNumber(''), 'to equal', 'Invalid phone number'))
+
+  it('returns "Invalid phone number" if an invalid phone number is provide', () =>
```
Comment (ranzwertig) [inline src/app/lib/validator.spec.js]:
```suggestion
  it('returns "Invalid phone number" if an invalid phone number is provided', () =>
```

File: src/app/component/country-select-field/_country-select-field.scss [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+  }
+
+  &__readMode {
+    height: $field-height;
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+
+    & > * {
+      margin-left: $space-base;
```
Comment (ranzwertig) [inline src/app/component/country-select-field/_country-select-field.scss]: should there be a margin left on the first child?

Comment (ranzwertig) [reply src/app/component/country-select-field/_country-select-field.scss]: PixelSnap 2019-07-26 at 10 21 52

Comment (dreampulse) [reply src/app/component/country-select-field/_country-select-field.scss]: The left margin here is on purpose to align with this here
image

**Comment by dreampulse** [comment]
> When I am ordering on behalf of a company, should the company name be added to the delivery address? Banause there is a country field too. Maybe I need the FAT ID but like to deliver to my provate address?

I think thats wired too. But they want to do it this way

File: src/app/component/country-select-field/_country-select-field.scss (hunk 13 above)
Comment (ranzwertig) [reply src/app/component/country-select-field/_country-select-field.scss]: hmm I think this looks ugly :)

File: src/app/component/country-select-field/_country-select-field.scss [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+    font-size: $font-size-s;
+  }
+
+  &__readMode {
+    height: $field-height;
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+
+    & > * {
```
Comment (ranzwertig) [inline src/app/component/country-select-field/_country-select-field.scss]: If the spacing stays I would use a padding in the `__readMode ` and make this a & > * + *

**Pull request merged**